### PR TITLE
Add boundary tests for JsonTreeWriter.endArray() error handling

### PR DESCRIPTION
**Pull request by lmj798**
## Summary
Add 3 boundary tests for `JsonTreeWriter.endArray()` to verify proper exception throwing when called in invalid states:
1. `testEndArrayOnEmptyStackThrows` - Verifies IllegalStateException when calling endArray() on empty stack
2. `testEndArrayWithPendingNameThrows` - Verifies IllegalStateException when a pending name exists (object not closed)
3. `testEndArrayWhenStackTopIsNotArrayThrows` - Verifies IllegalStateException when stack top is an object, not an array
## Why
Current tests cover:
- Normal array/object begin/end operations
- Close after write
- Premature close
- Null handling
- NaN/Infinity handling
However, they don't explicitly test `endArray()` error conditions for invalid state detection. These boundary cases ensure proper state validation and prevent regressions from state management changes.
## Verification
mvn -Dtest=JsonTreeWriterTest test
Result: Tests run: 25, Failures: 0, Errors: 0, Skipped: 0 - BUILD SUCCESS

### CODE REVIEW

File: gson/src/test/java/com/google/gson/internal/bind/JsonTreeWriterTest.java [hunk 1]
```diff
@@ -100,7 +100,7 @@ public void testPrematureClose() throws Exception {
     JsonTreeWriter writer = new JsonTreeWriter();
     writer.setStrictness(Strictness.LENIENT);
     writer.beginArray();
-    var e = assertThrows(IOException.class, () -> writer.close());
+    IOException e = assertThrows(IOException.class, () -> writer.close());
```
Comment (eamonnmcmanus) [inline gson/src/test/java/com/google/gson/internal/bind/JsonTreeWriterTest.java]: Is there a reason for this change?

**Review by eamonnmcmanus — CHANGES_REQUESTED** [review]
This seems like a reasonable change. Just one question.

File: gson/src/test/java/com/google/gson/internal/bind/JsonTreeWriterTest.java (hunk 1 above)
Comment (lmj798) [reply gson/src/test/java/com/google/gson/internal/bind/JsonTreeWriterTest.java]: I changed it back to explicit typing to keep the coding style consistent with our main source code (which is Java 8). Mixing var in tests while strictly using explicit types in production code can be inconsistent.

Comment (Marcono1234) [reply gson/src/test/java/com/google/gson/internal/bind/JsonTreeWriterTest.java]: The test code intentionally uses a higher target Java version (11) to make it more convenient and succinct to write. So this inconsistency might not be a problem?

**Comment by Marcono1234** [comment]
Could you please move your last commit (https://github.com/google/gson/pull/2988/changes/073d0f5f2fb364e572e18793c1f2ac1077cec9c0) to a separate PR? It is unrelated to `JsonTreeWriter.endArray()`.

**Comment by lmj798** [comment]
> Could you please move your last commit ([073d0f5](https://github.com/google/gson/commit/073d0f5f2fb364e572e18793c1f2ac1077cec9c0)) to a separate PR? It is unrelated to `JsonTreeWriter.endArray()`.

Sorry about that, and thank you for catching it.
You’re absolutely right that commit 073d0f5 was unrelated. I’ve moved it to a separate PR and force-pushed this branch, so this PR now only contains the JsonTreeWriter.endArray() changes.
